src/rules/sortImportsRule.ts

```diff
@@ -0,0 +1,246 @@
+    // Cheesing array compare
+    const imports: string[] = node.elements.map(e => this.caseConverter(e.getText()));
+    const importReduction = imports.reduce((prev, current) => prev + current);
+    const sortedImports = imports.sort();
+    const sortedReduction = sortedImports.reduce((prev, current) => prev + current);
+
+    if (importReduction !== sortedReduction) {
+      this.addFailureAtNode(
+        node,
+        'Member imports must be sorted alphabetically.');
+    }
+  }
+
+  private _validateOrder(node: ts.ImportDeclaration | ts.ImportEqualsDeclaration) {
+    const importData = this._determineImportType(node);
+
+    // See if the import type is still available
+    const index = this.expectedOrder.indexOf(importData.memberSyntaxType, this.currentImportIndex);
+    if (index !== -1) {
+      if (this.expectedOrder[this.currentImportIndex] !== importData.memberSyntaxType) {
+        this.currentImportIndex = index;
+        this.currentSortValue = {
+          sortValue: this.caseConverter(importData.sortValue),
+          originalValue: importData.sortValue};
+      } else if (this.currentSortValue.sortValue > this.caseConverter(importData.sortValue)) {
+        this.addFailureAtNode(
+          node,
+          `All imports of the same type must be sorted alphabetically. "${importData.sortValue}" must come before "${this.currentSortValue.originalValue}"`);
+      } else {
+        this.currentSortValue = {
+          sortValue: this.caseConverter(importData.sortValue),
+          originalValue: importData.sortValue};
+      }
+    } else {
+      const currentSyntaxType = MemberSyntaxType[importData.memberSyntaxType];
+      const previousSyntaxType = MemberSyntaxType[this.expectedOrder[this.currentImportIndex]];
+      this.addFailureAtNode(
+        node,
+        `All imports of type "${currentSyntaxType}" must occur before all imports of type "${previousSyntaxType}"`);
+    }
+  }
+
+  private _determineImportType(node: ts.ImportDeclaration | ts.ImportEqualsDeclaration): ImportMetadata {
+    const nodeText = node.getFullText();
+
+    if (node.kind === ts.SyntaxKind.ImportEqualsDeclaration) {
+      const aliasMatch = /\bimport\s+(\w+)\s*=.+/g.exec(nodeText);
+      return {
+        memberSyntaxType: MemberSyntaxType.Alias,
+        sortValue: aliasMatch[1]
+      };
+    } else {
+      const singleMatch = /\bimport\s+({?([^,{}\*]+?)}?)\s*from\s+[\'"](?:[^"\']+)["\']/g.exec(nodeText);
+      const multipleMatch = /\bimport\s*{?\s*([^{}\'",]+?)\s*,(?:\s*.+\s*,\s*)*\s*.+\s*}?\s*from\s+[\'"](?:[^"\']+)["\']/g.exec(nodeText);
+      const noneMatch = /\bimport\s+[\'"]([^"\']+)["\']/g.exec(nodeText);
+      const allMatch = /\bimport\s+\*\s+as\s+(.+)\s+from\s+[\'"](?:[^"\']+)["\']/g.exec(nodeText);
+
+      if (singleMatch !== null) {
+        return {
+          memberSyntaxType: MemberSyntaxType.Single,
+          sortValue: singleMatch[1]
+        };
+      } else if (multipleMatch !== null) {
```

Comment:
Since you are returning within each if statement you can avoid writing `else if`. Just make them `if` statements since you won't reach them due to the return statement.